lighthouse-core/gather/gatherers/image-elements.js

```diff
@@ -142,6 +167,72 @@ function determineNaturalSize(url) {
   });
 }
 
+/**
+ * @param {LH.Crdp.CSS.CSSStyle} [style]
+ * @param {string} property
+ * @return {boolean}
+ */
+function hasSizeDeclaration(style, property) {
+  return !!style && !!style.cssProperties.find(({name}) => name === property);
+}
+
+/**
+ * Finds the most specific directly matched CSS font-size rule from the list.
+ *
+ * @param {Array<LH.Crdp.CSS.RuleMatch>} [matchedCSSRules]
+ * @param {string} property
+ * @returns {string}
+ */
+function findMostSpecificMatchedCSSRule(matchedCSSRules = [], property) {
```

Comment:
wow you already got the CSS sizing in here @lemcardenas that's amazing!

adding the extra tests for this case will make this PR get pretty large though and makes the logic in ImageElements a lot heftier that we'll want to smoketest, maybe we could split this functionality out into a separate PR? to limit the churn to the audit I would say leave the `cssWidth`/`cssHeight` properties *optional* in the artifact typedef so you can still use them and test in the audit in this PR.

how does that sound?

Reply:
That sounds good!
To clarify, this is what I plan to do based on this guidance:

1. make `cssWidth`/`cssHeight` optional
2. keep css logic in `no-unsized-images.js`
3. add css testing in `no-unsized-images-test.js`
4. remove the css logic in `image-elements.js`for this PR
5. create a new branch/PR with the css logic in `image-elements.js`
6. add a smoketest for the branch in 5. 

Does this sound in line with what you recommended?

Reply:
+1

as a reviewer I also need to brush off the ol' `font-size` and re-familiarize myself with all this.

Reply:
yup great plan @lemcardenas !